metadata/metadata.go

```diff
@@ -10,47 +10,60 @@ import (
 const AppMetadataLabel = "io.buildpacks.lifecycle.metadata"
 
 type AppImageMetadata struct {
-	App        AppMetadata         `json:"app"`
-	Config     ConfigMetadata      `json:"config"`
-	Launcher   LauncherMetadata    `json:"launcher"`
-	Buildpacks []BuildpackMetadata `json:"buildpacks"`
-	RunImage   RunImageMetadata    `json:"runImage"`
-	Stack      StackMetadata       `json:"stack"`
+	App        AppMetadata         `json:"app" toml:"app"`
+	Config     ConfigMetadata      `json:"config" toml:"config"`
+	Launcher   LauncherMetadata    `json:"launcher" toml:"launcher"`
+	Buildpacks []BuildpackMetadata `json:"buildpacks" toml:"buildpacks"`
+	RunImage   RunImageMetadata    `json:"runImage" toml:"run-image"`
+	Stack      StackMetadata       `json:"stack" toml:"stack"`
 }
 
 type AppMetadata struct {
-	SHA string `json:"sha"`
+	SHA string `json:"sha" toml:"sha"`
+}
+
+type AnalyzedMetadata struct {
+	Repository string           `toml:"repository"`
+	Digest     string           `toml:"digest"`
+	Metadata   AppImageMetadata `toml:"metadata"`
+}
+
+func (a AnalyzedMetadata) FullName() string {
+	if a.Digest == "" {
+		return a.Repository
+	}
+	return a.Repository + "@" + a.Digest
 }
 
 type ConfigMetadata struct {
-	SHA string `json:"sha"`
+	SHA string `json:"sha" toml:"sha"`
 }
 
 type LauncherMetadata struct {
-	SHA string `json:"sha"`
+	SHA string `json:"sha" toml:"sha"`
 }
 
 type BuildpackMetadata struct {
-	ID      string                   `json:"key"`
-	Version string                   `json:"version"`
-	Layers  map[string]LayerMetadata `json:"layers"`
+	ID      string                   `json:"key" toml:"key"`
+	Version string                   `json:"version" toml:"version"`
+	Layers  map[string]LayerMetadata `json:"layers" toml:"layers"`
 }
 
 type LayerMetadata struct {
-	SHA    string      `json:"sha" toml:"-"`
+	SHA    string      `json:"sha" toml:"sha"`
```

Comment:
We don't want to write the layer sha to the layer metadata files in the layers dir. This might cause issues.